packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -10,6 +10,8 @@ import '../globals.dart';
 
 typedef String StringConverter(String string);
 
+final Map<String, String> childEnvironmentOverrides = new Map<String, String>();
```

Comment:
docs say that if `incldeParentEnvironment` is true (default), it will include the parent process' environment, giving precedence to the overrides for name conflicts -- so we should be good.